examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoFeedTableViewController.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  PhotoFeedTableViewController.swift
+//  ASDKgram-Swift
+//
+//  Created by Calum Harris on 06/01/2017.
+//
+//  Copyright (c) 2014-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+//  FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+//   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+//  CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import UIKit
+
+class PhotoFeedTableViewController: UITableViewController {
+
+	var activityIndicator: UIActivityIndicatorView!
+	var photoFeed: PhotoFeedModel
+
+	init() {
+		photoFeed = PhotoFeedModel(initWithPhotoFeedModelType: .photoFeedModelTypePopular, requiredImageSize: screenSizeForWidth)
+		super.init(nibName: nil, bundle: nil)
+		self.navigationItem.title = "UIKit"
+
+	}
+
+	required init?(coder aDecoder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	override func viewDidLoad() {
+
+		activityIndicatorConfig()
+		tableViewProperties()
+		fetchNewBatch()
+		navigationController?.hidesBarsOnSwipe = true
+	}
+
+	func fetchNewBatch() {
+
+		activityIndicator.startAnimating()
+		photoFeed.updateNewBatchOfPopularPhotos() { additions, connectionStatus in
+			switch connectionStatus {
+			case .connected:
+				self.activityIndicator.stopAnimating()
+				self.addRowsIntoTableView(newPhotoCount: additions)
+			case .noConnection:
+				self.activityIndicator.stopAnimating()
+				break
+			}
+		}
+	}
+
+	var screenSizeForWidth: CGSize = {
+		let screenRect = UIScreen.main.bounds
+		let screenScale = UIScreen.main.scale
+		return CGSize(width: screenRect.size.width * screenScale, height: screenRect.size.width * screenScale)
+	}()
+
+	// helper functions
+
+	func activityIndicatorConfig() {
+		let activityIndicator = UIActivityIndicatorView(activityIndicatorStyle: .gray)
```

Comment:
Let's name this `setupActivityIndicator()`, and the method below `configureTableView()`